Transaction.proto

```diff
@@ -0,0 +1,121 @@
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Network.Protocol.Protobuffs <https://github.com/catalyst-network/protocol-protobuffs>
+*
+* Catalyst.Network.Protocol.Protobuffs is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+* 
+* Catalyst.Network.Protocol.Protobuffs is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+* 
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Network.Protocol.Protobuffs If not, see <https://www.gnu.org/licenses/>.
+*/
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package Catalyst.Protocol.Transaction;
+
+/**
+ * Transaction
+ *
+ * Version: 1 for non-confidential transaction, 2 for confidential transaction
+ * TimeStamp:
+ * TransactionFees: 8 bytes, clear text, fees * 10^12 - always positive
+ * LockTime: 32 bits
+ * STEntries: 0 field for confidential transaction
+ * CFEntries: 0 field for standard transaction
+ * Signature:
+ * EntryRangeProof: 0 field for standard transaction
+ */
+message Transaction {
+    uint32 Version = 1 [default = 1];
+    uint64 TimeStamp = 2;
+    uint64 TransactionFees = 3;
+    uint64 LockTime = 4;
+    repeated STTransactionEntry STEntries = 5;
+    repeated CFTransactionEntry CFEntries = 6;
+    message TransactionSignature Signature = 7; 
+    repeated message EntryRangeProof = 8;
+}
+
+/**
+ * STTransactionEntry
+ *
+ * PubKey: 32 bytes, (255 bits or 256 with last bit to 0) public key (account address derived from the public key)
+ * Amount: max 8 bytes (positive or negative)
+ */
+message STTransactionEntry {
+    bytes PubKey = 1;
+    sint64 Amount = 2;
+}
+ 
+/**
+ * CFTransactionEntry
+ *
+ * PubKey: 32 bytes, account address derived from the public key
+ * PedersenCommitment: 32bytes
+ */
+message CFTransactionEntry { 
+    bytes PubKey = 1;
+    bytes CommPedersenCommitmentitment = 2;
+}
+
+/**
+ * TransactionSignature
+ *
+ * Signature: 32 bytes, 1 aggregated signature per transaction  (e: signature; B: schnorr component)
+ * Challenge: 32 bytes
+ */
+message TransactionSignature {
+    bytes Signature = 1;
+    bytes Challenge = 2;
+}
+
+/**
+ * EntryRangeProof
+ *
+ * V:
+ * A:
+ * T1:
+ * T2:
+ * TAU:
+ * MU:
+ * L:
+ * R:
+ * APrime0:
+ * BPrime0:
+ */
+message EntryRangeProof {
```

Comment:
Can we try to get some more meaningful names for these variables ?

Reply:
Not sure how more meaningful it can get, these are EC points and elements in the Zp field that compose a bulletproof structure. We can relate these to the equations in the bulletproof paper but ultimately, to understand them one would have to read the paper.

Reply:
Is there an explanation for example what V is?

Reply:
Thanks @franssl